pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -916,7 +932,7 @@ func handleCreate(ctx *domainContext, key string, config *types.DomainConfig) {
 		DomainName:         name,
 		AppNum:             config.AppNum,
 		VifList:            config.VifList,
-		VirtualizationMode: config.VirtualizationMode,
+		VirtualizationMode: virtualizationMode,
 		EnableVnc:          config.EnableVnc,
 		VncDisplay:         config.VncDisplay,
 		VncPasswd:          config.VncPasswd,
```

Comment:
Its better to spin this off into a routine like:
        types.DomainStatusFromConfig(config)

This way - such default values can be controlled better..

Reply:
Yeah but it wouldn't help in this case since we also do this for handleUpdate, in which case we don't just replace the status but only update a few fields.

Reply:
Applied your other suggestion below.